app-libs/parentchain-interface/src/lib.rs

```diff
@@ -54,3 +65,18 @@ pub fn decode_and_log_error<V: Decode>(encoded: &mut &[u8]) -> Option<V> {
 		},
 	}
 }
+
+/// a trait to aggreagate all traits we need in the service
+pub trait ParentchainApiTrait:
+	AccountApi<AccountId = AccountId, Balance = Balance, Index = Index>
+	+ GetBalance<Balance = Balance>
+	+ GetStorage
+	+ SystemApi<Properties = Properties>
+	+ BalancesExtrinsics<Address = MultiAddress<AccountId, Index>, Balance = Balance>
+	+ ChainApi
+	+ GetAccountInformation<Index = Index>
+	+ SubscribeEvents
+	+ GetTransactionPayment<Balance = Balance>
+	+ SubmitAndWatch<Hash = Hash> //+ GetRuntimeVersionAt<Block<Header, UncheckedExtrinsic<A>>>
+{
+}
```

Comment:
this supertrait shall simplify using generic NodeApis, defined per Parentchain in app-libs

Reply:
I am a bit unsure about this trait. Even if we have it, I would only make it a trait bound in as few places as possible, as 99% of the times we only need part of it.